src/biokbase/narrative/jobs/job.py

```diff
@@ -300,35 +284,26 @@ def parameters(self):
                     f"Unable to fetch parameters for job {self.job_id} - {e}"
                 )
 
-    def update_state(self, state):
+    def update_state(self, state: dict) -> dict:
         """
         given a state data structure (as emitted by ee2), update the stored state in the job object
         """
-        if not state:
-            return
+        if state:
 
-        if "job_id" in state and state["job_id"] != self.job_id:
-            raise ValueError(
-                f"Job ID mismatch in update_state: job ID: {self.job_id}; state ID: {state['job_id']}"
-            )
-
-        state = copy.deepcopy(state)
-        if self._last_state is None:
-            self._last_state = state
-        else:
-            # TODO: implement batch job code updates here
-            # TODO: update state structure-wise
-            self._last_state.update(state)
+            if "job_id" in state and state["job_id"] != self.job_id:
+                raise ValueError(
+                    f"Job ID mismatch in update_state: job ID: {self.job_id}; state ID: {state['job_id']}"
+                )
 
-        return copy.deepcopy(self._last_state)
+            state = copy.deepcopy(state)
+            if self._acc_state is None:
+                self._acc_state = state
+            else:
+                self._acc_state.update(state)
 
-    def clear_state(self):
-        """
-        reset the internal job state by removing the stored state
-        """
-        self._last_state = None
```

Comment:
Only used in tests, unknown purpose, and didn't agree so well with the concept of "accumulated state"